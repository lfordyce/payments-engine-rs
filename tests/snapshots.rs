use assert_cmd::prelude::*;
use std::process::Command;
use std::str;

#[test]
fn basic() -> Result<(), Box<dyn std::error::Error>> {
    let mut cmd = Command::cargo_bin("payments-engine-rs")?;
    cmd.arg("./etc/basic.csv");
    let stdout = String::from_utf8(cmd.assert().success().get_output().stdout.clone())?;

    insta::assert_snapshot!(stdout);

    Ok(())
}

#[test]
fn trigger_locked() -> Result<(), Box<dyn std::error::Error>> {
    let mut cmd = Command::cargo_bin("payments-engine-rs")?;
    cmd.arg("./etc/locked.csv");
    let stdout = String::from_utf8(cmd.assert().success().get_output().stdout.clone())?;

    insta::assert_snapshot!(stdout);

    Ok(())
}

#[test]
fn dispute_deposit() -> Result<(), Box<dyn std::error::Error>> {
    let mut cmd = Command::cargo_bin("payments-engine-rs")?;
    cmd.arg("./etc/dispute_deposit.csv");
    let stdout = String::from_utf8(cmd.assert().success().get_output().stdout.clone())?;

    insta::assert_snapshot!(stdout);

    Ok(())
}

#[test]
fn deposit_chargeback() -> Result<(), Box<dyn std::error::Error>> {
    let mut cmd = Command::cargo_bin("payments-engine-rs")?;
    cmd.arg("./etc/deposit_chargeback.csv");
    let stdout = String::from_utf8(cmd.assert().success().get_output().stdout.clone())?;

    insta::assert_snapshot!(stdout);

    Ok(())
}

#[test]
fn funds_held() -> Result<(), Box<dyn std::error::Error>> {
    let mut cmd = Command::cargo_bin("payments-engine-rs")?;
    cmd.arg("./etc/funds_held.csv");
    let stdout = String::from_utf8(cmd.assert().success().get_output().stdout.clone())?;

    insta::assert_snapshot!(stdout);

    Ok(())
}

#[test]
fn smoke_test() -> Result<(), Box<dyn std::error::Error>> {
    let mut cmd = Command::cargo_bin("payments-engine-rs")?;
    cmd.arg("./etc/smoketest.csv");
    let stdout = String::from_utf8(cmd.assert().success().get_output().stdout.clone())?;

    insta::assert_snapshot!(stdout);

    Ok(())
}
