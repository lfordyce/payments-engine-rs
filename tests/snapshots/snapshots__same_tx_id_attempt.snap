---
source: tests/snapshots.rs
expression: stdout
---
client,available,held,total,locked
1,135,0,135,false
